### PR TITLE
Add automatic stylesheet enqueuing

### PR DESCRIPTION
**Pull request by lukecarbis**
This is a _much_ neater way to do block styles.

On the frontend, Block Lab will enqueue the first of these stylesheets, for any blocks used on the current page:
- `{child theme directory}/blocks/css/block-{block name}.css`
- `{child theme directory}/blocks/block-{block name}.css`
- `{parent theme directory}/blocks/css/block-{block name}.css`
- `{parent theme directory}/blocks/block-{block name}.css`

In the editor, Block Lab will enqueue the first of these stylesheets, for _all blocks_ (since any one of them could be inserted at any point):
- `{child theme directory}/blocks/css/preview-{block name}.css`
- `{child theme directory}/blocks/preview-{block name}.css`
- `{child theme directory}/blocks/css/block-{block name}.css`
- `{child theme directory}/blocks/block-{block name}.css`
- `{parent theme directory}/blocks/css/preview-{block name}.css`
- `{parent theme directory}/blocks/preview-{block name}.css`
- `{parent theme directory}/blocks/css/block-{block name}.css`
- `{parent theme directory}/blocks/block-{block name}.css`

Please test each possibility!

### CODE REVIEW

**Comment by RobStino** [comment]
:slow_clap:
This is so cool man! Such an elegant solution (to be tested) to styling for blocks!
Bonus points for the conditional enqueueing on the frontend!

**Comment by lukecarbis** [comment]
Do not merge until I've written unit tests, which I'll do once #233 is merged.

**Comment by lukecarbis** [comment]
Okay, @kienstra ready for you to review and merge.

**Comment by RobStino** [comment]
Tested all variations of file locations and everything works great!

File: php/blocks/class-loader.php
Comment (kienstra) [inline php/blocks/class-loader.php]: Would this method fit well in the `Loader` class, instead of `php/helpers.php`? Though maybe it'll later be used in other classes, I'm not sure.

File: tests/php/test-helpers.php
Comment (kienstra) [inline tests/php/test-helpers.php]: Sometimes I'll use `$wp_styles = wp_styles();`, but the effect is the same.

**Review by kienstra — DISMISSED** [review]
**Approved**

Hi @lukecarbis,
Great idea to enqueue styling for the blocks. The styling appeared as expected:

styling-appears

There's a question below, but not a blocker to merging if you're satisfied.

**Review by RobStino — DISMISSED** [review]
Tested with all file variations and all worked as expected. Great work!

File: tests/php/test-helpers.php
Comment (lukecarbis) [reply tests/php/test-helpers.php]: Updated.

File: php/blocks/class-loader.php
Comment (lukecarbis) [reply php/blocks/class-loader.php]: Great idea – done in 6148de0.

**Comment by lukecarbis** [comment]
@kienstra Could you please review 6148de0?

**Review by kienstra — APPROVED** [review]
**Approved**

Hi @lukecarbis,
Sorry for the delay. This looks good, and the block stylesheet still appears as expected:

stylesheet-enqueued

**Pull request merged**
